client/packages/core/src/queryTypes.ts

```diff
@@ -84,9 +139,34 @@ type $Option<Fields extends string[]> = {
   };
 };
 
+// Helper type to check if an entity has index signature (unknown schema)
+type HasIndexSignature<T> = string extends keyof T ? true : false;
+
+// Typed version that supports dot notation for nested queries
+type $SchemaOption<
+  S extends IContainEntitiesAndLinks<any, any>,
+  K extends keyof S['entities'],
+> = {
+  $?: {
+    where?: WhereClause<S, K>;
+    order?: Order;
+    limit?: number;
+    last?: number;
+    first?: number;
+    offset?: number;
+    after?: Cursor;
+    before?: Cursor;
+    fields?: HasIndexSignature<InstaQLEntity<S, K>> extends true
```

Comment:
Can we try to remove this check? 

imo if the index signature exists here, wouldn't InstaQLFields end up basically returning string[]? 

The idea being we should do our best to not special-case against a specific kind of schema